flask/flask-launcher/src/main/java/net/corda/flask/launcher/JavaProcessBuilder.java

```diff
@@ -0,0 +1,183 @@
+                }
+                if (process[0].isAlive()) {
+                    process[0].destroyForcibly();
+                }
+            }
+        }));
+        process[0] = build().inheritIO().start();
+        return process[0].waitFor();
+    }
+
+    @SneakyThrows
+    public ProcessBuilder build() {
+        ArrayList<String> cmd = new ArrayList<>();
+        Path javaBin = Paths.get(javaHome, "bin", "java");
+        cmd.add(javaBin.toString());
+        cmd.addAll(jvmArgs);
+        if(!classpath.isEmpty()) {
+            cmd.add("-cp");
+            cmd.add(classpath.stream().collect(Collectors.joining(PATH_SEPARATOR)));
+        }
+        for(Map.Entry<Object, Object> entry : properties.entrySet()) {
+            cmd.add(String.format("-D%s=%s", entry.getKey(), entry.getValue()));
+        }
+        for(JavaAgent javaAgent : javaAgents) {
+            StringBuilder sb = new StringBuilder("-javaagent:" + javaAgent.jar.toString());
+            String agentArguments = javaAgent.args;
+            if(agentArguments != null) {
+                sb.append('=');
+                sb.append(agentArguments);
+            }
+            cmd.add(sb.toString());
+        }
+        if(executableJar != null) {
+            cmd.add("-jar");
+            cmd.add(executableJar.toString());
+        } else if(mainClassName != null) {
+            cmd.add(mainClassName);
+        } else {
+            throw new IllegalArgumentException(
+                    "Either a main class or the path to an executable jar file have to be specified");
+        }
+        cmd.addAll(cliArgs);
+
+        int cmdLength = 0;
+        for(String part : cmd) {
+            cmdLength += part.length();
+        }
+        //Add space between arguments
+        cmdLength += cmd.size() - 1;
+
+        if(log.isDebugEnabled()) {
+            log.debug("Spawning new process with command line: [{}]",
+                    cmd.stream().map(s -> "\"" + s + "\"").collect(Collectors.joining(", ")));
+        }
+        if(cmdLength < COMMAND_LINE_MAX_SIZE || cmd.size() == 1) {
+            return new ProcessBuilder(cmd);
+        } else {
+            Path argumentFile = Files.createTempFile(PROCESS_BUILDER_PREFIX, ".arg");
+            Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        Files.delete(argumentFile);
+                    } catch (IOException ioe) {
+                        ioe.printStackTrace();
+                    }
+                }
+            }));
+            log.trace("Using Java argument file '{}'", argumentFile);
+            try(Writer writer = Files.newBufferedWriter(argumentFile)) {
+                writer.write(generateArgumentFileString(cmd.subList(1, cmd.size())));
+            }
+            return new ProcessBuilder(Arrays.asList(cmd.get(0), "@" + argumentFile.toString()));
```

Comment:
You don't need to use `Arrays.asList()` here because `ProcessBuilder` also has a `ProcessBuilder(String... args)` constructor.